test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReportGenerator.cs

```diff
@@ -0,0 +1,303 @@
+                //                              = sum(x^2) / (N - 1) - mean^2
+                variancePeriodInMilisec = (totalSpanSquareInMilisec / Math.Max(orderedCompletedPeriods.Count() - 1, 1)) - Math.Pow(meanPeriod.TotalMilliseconds, 2);
+            }
+
+            return new EdgeHubRestartDirectMethodReport(
+                this.TrackingId,
+                this.TestReportType.ToString(),
+                passedDirectMethodCount,
+                this.SenderSource,
+                this.ReceiverSource,
+                senderResultCount,
+                receiverResultCount,
+                this.completedStatusHistogram,
+                minPeriod,
+                maxPeriod,
+                medianPeriod,
+                meanPeriod,
+                variancePeriodInMilisec);
+        }
+
+        async Task<(ulong resultCount, bool isNotEmpty, long sequenceNum)> IncrementSequenceNumberAsync(
+            bool isNotEmpty,
+            MoveNextResultAsync MoveNextResultAsync,
+            long targetSequenceNumber,
+            ulong resultCount)
+        {
+            long seqNum = 0;
+
+            while (isNotEmpty && (seqNum < targetSequenceNumber))
+            {
+                (resultCount, isNotEmpty, seqNum) = await MoveNextResultAsync(resultCount);
+            }
+
+            return (resultCount: resultCount, isNotEmpty: isNotEmpty, sequenceNum: seqNum);
+        }
+
+        void ValidateResult(
+            TestOperationResult result,
+            string expectedSource,
+            string testOperationResultType)
+        {
+            if (!result.Source.Equals(expectedSource, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new InvalidDataException($"Result source is '{result.Source}' but expected should be '{expectedSource}'.");
+            }
+
+            if (!result.Type.Equals(testOperationResultType, StringComparison.Ordinal))
+            {
+                throw new InvalidDataException($"Result type is '{result.Type}' but expected should be '{testOperationResultType}'.");
+            }
+        }
+
+        //////////////////////////////////////////////////////////////// HELPER LAND
+        void AddEntryToCompletedStatusHistogram(EdgeHubRestartDirectMethodResult senderResult)
+        {
+            HttpStatusCode completedStatus = senderResult.DirectMethodCompletedStatusCode;
+            TimeSpan completedPeriod = senderResult.DirectMethodCompletedTime - senderResult.EdgeHubRestartedTime;
+            // Try to allocate the list if it is the first time HttpStatusCode shows up
+            this.completedStatusHistogram.TryAdd(completedStatus, new List<TimeSpan>());
+            this.completedStatusHistogram[completedStatus].Add(completedPeriod);
+        }
+
+        long ConvertStringToLong(string result)
+        {
+            long seqNum;
+            long.TryParse(result.Split(';').LastOrDefault(), out seqNum);
```

Comment:
seq number should be the third element after split; please check if splited result has 3 element exactly and return third element.  Don't use LastOrDefault().